### DISCUSSION TITLE
Indicator of launching action

### DISCUSSION THREAD

**Discussion by HansenChristoffer** [lutris/lutris#5081] [Ideas]
## Background
Hello!

After using Lutris for quite some time, there is one thing I've always missed. This being some kind of indicator on whether you're in progress of launching something after you've double-clicked on an application.
I'll write two scenarios, one how it is today and another how I could imagine it being. Obviously, the actual indicator could be anything really, as long it is expressive and noticeable enough.

I can see the risk that perhaps my Lutris acts different because of OS (Manjaro Linux x86_64), DE (Xfce 4.18) and possibly WM (Xfwm4). Do please inform me if that would be the case.

### How it is today
1. Run Lutris.
2. Click Platforms -> Windows.
3. Click on the application you would like to start, which should highlight the application.
4. Double-click on the application. Nothing happens.
5. Eventually the application appears on the screen.

### How I imagine it
1. Run Lutris.
2. Press Platforms -> Windows.
3. Click on the application you would like to start, which should highlight the application.
4. Double-click on application. Your highlight quickly disappears and reappears back again, as an indicator that an action has occurred upon this application.
5. Eventually the application appears on the screen.

**Comment by strycore** [comment]
Like for example having the Play button switch to saying "Launching" in an inactive state before switching to "Stop"?

Because we've had that for years.

**Reply by HansenChristoffer** [reply]
After this comment I decided to go through my list of games, I also added couple of new ones to test out.

It seems like the "Button" changing state only occurred for me on the Steam games, that is ran with Proton. I tried to start Battle.net and it does not change the button state at all. I tried to start project ascension, no button state change.
battle.net being the application I've been using the most, hence why I did not know of the buttons state changing.

I personally still think the indication directly on the action of double clicking an application would be more expressive and straight forward. Not necessarily an indicator that the game is launching but at least that you as a user has done an "action".

I guess it is a bug that certain applications does not trigger button state(s)?

Correction: It seems like a "launcher" does not trigger it but if you enter a game from said launcher, the button state changes.

**Reply by strycore** [reply]
If nothing in the configuration has changed then the prelaunch step can be too fast to show a "Launching" step on the button.

It would be nice to have some kind of animation on the game cover, only problem, I have no idea how to do animations in Gtk.

**Reply by HansenChristoffer** [reply]
The GUI is a bit flaky on my computer. Whether it is Lutris or any combination of OS, DE or WM I do not know. But sometimes the buttons state gets changed instantly and graphically rendered instantly. Other times it never gets updated until you make it re-render the "page".

I've never done any GTK programming myself, so I am not much of help. I will take a look but am not promising anything.
